xline/src/storage/kv_store.rs

```diff
@@ -496,6 +498,13 @@ where
 
     /// Handle `RangeRequest`
     fn handle_range_request(&self, req: &RangeRequest) -> Result<RangeResponse, ExecuteError> {
+        req.validation()?;
+        Self::check_revision(
+            &UniRequest::RequestRange(req.clone()),
```

Comment:
It seems that when we introduce `UniRequest` in `check_revision`, we have to clone the request here.
Can we optimize it? Or we could add a TODO here.

Reply:
I refactored the revison check code, now it only takes a reference